avm/res/container-service/managed-cluster/main.bicep

```diff
@@ -739,15 +742,15 @@ resource managedCluster 'Microsoft.ContainerService/managedClusters@2024-03-02-p
       }
     }
     publicNetworkAccess: publicNetworkAccess
-    aadProfile: {
+    aadProfile: enableAad ? {
```

Comment:
If there are any required parameters, would it not maybe be simpler to change this condition do something like `aadProfile: !empty(aadProfileClientAppID) ? {` ?

Mind you, I have no clue if any of those parameter is required, so this is just an example.

An alternative, by the way, would be to add a user defined type for the `aadProfile` that is used by a corresponding parameter: `param aadProfile aadProfileType?` and would contain all the below properties. If implemented, the above suggested check could be changed to something like `aadProfile: !empty(aadProfile) ? {` 

Just some ideas 😏

Reply:
do we need this? aadProfileManaged specifies whether to enabled managed AAD integration already.

Reply:
@PixelRobots `aadProfileManaged` does control whether managed AAD integration is enabled, but it does not disable AAD integration itself. It only affects whether Azure managed identities are used.

Specifically, when `aadProfileManaged` is set to false, it simply means that the AAD integration automatically managed by Azure is not used, but the user is required to provide relevant information about the AAD app (such as `clientAppID`, `serverAppID`, `serverAppSecret`, etc.). At this time, AAD integration is still enabled, but it is manually configured instead of being managed by Azure.

Reply:
Could we use the `aadProfileManaged` as the paramater rather than creating a new one?

Also, we need to ensure that local accounts are not disabled. so `disableLocalAccounts` needs to be set as `false` too. Otherwise, it will fail deployment.

Maybe we need a new test file for non aad cluster also.

Reply:
@PixelRobots, Thanks for your feedback! To clarify, the value of this parameter `aadProfileManaged` cannot be used as a criterion for whether AAD integration is enabled.

 Because when `aadProfileManaged` is set to `false`, AAD integration itself is still `enabled`, just manually configured. When `aadProfileManaged` is set to `true`, AAD integration is automatically managed by Azure.

Reply:
@PixelRobots I have updated the code to add a user defined type for `aadProfile` as @AlexanderSehr suggested and also added a new e2e/test file for non aad cluster.

Reply:
Nice. Looks super clean. @JPEasier can you approve this pull request?